data/model-db/generate_historical_data.R

```diff
@@ -0,0 +1,778 @@
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_carcass.csv",
+               overwrite = TRUE)
+carcass_raw <- read_csv("data/standard-format-data/standard_carcass.csv")
+
+carcass_reach <- carcass_raw |> 
+  group_by(stream) |> 
+  distinct(reach)
+# carcass_estimates -------------------------------------------------------
+# TODO need to confirm the run and adipose_clipped for this data
+gcs_get_object(object_name = "standard-format-data/standard_carcass_cjs_estimate.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_carcass_cjs_estimate.csv",
+               overwrite = TRUE)
+carcass_estimates_raw <- read_csv("data/standard-format-data/standard_carcass_cjs_estimate.csv")
+
+carcass_estimates <- carcass_estimates_raw |> 
+  rename(carcass_estimate = spawner_abundance_estimate,
+         upper_bound_estimate = upper,
+         lower_bound_estimate = lower,
+         confidence_level = confidence_interval) |> 
+  # this is so we can join to survey_location
+  mutate(reach = NA) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |>
+  # need to confirm the run and adipose clipped
+  mutate(run = "spring",
+         adipose_clipped = F) |> 
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description) 
+
+try(if(any((unique(carcass_estimates$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(carcass_estimates$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(carcass_estimates$year)))
+  stop("Missing Year! Please fix!"))
+
+gcs_upload(carcass_estimates,
+           object_function = f,
+           type = "csv",
+           name = "model-db/carcass_estimates.csv",
+           predefinedAcl = "bucketLevel")
+
+# bulk carcass ------------------------------------------------------------
+# currently not going to use this table. originally designed to separate
+# out bulk counts from individual estimates
+
+# daily_redd --------------------------------------------------------------
+# TODO look into the unknow date entries for Feather River
+gcs_get_object(object_name = "standard-format-data/standard_daily_redd.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_daily_redd.csv",
+               overwrite = TRUE)
+daily_redd_raw <- read_csv("data/standard-format-data/standard_daily_redd.csv")
+
+daily_redd <- daily_redd_raw |> 
+  # remove the 100 or entries from feather river where date is unknown
+  # select for chinook 
+  filter(!is.na(date), species %in% c("chinook", "not recorded", "unknown")) |> 
+  select(date, latitude, longitude, reach, redd_id, age, age_index,
```

Comment:
We have an `age_index`column in this table that is not present in the ERD. Confirming that we want this column.

Reply:
Decided to remove age_index from the db